src/dotnet-interactive/Connection/StdIoKernelConnector.cs

```diff
@@ -13,40 +14,73 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+
 using Microsoft.DotNet.Interactive.Commands;
 using Microsoft.DotNet.Interactive.Connection;
 using Microsoft.DotNet.Interactive.Events;
 using Microsoft.DotNet.Interactive.Utility;
+
+using Pocket;
+
+using static Pocket.Logger<Microsoft.DotNet.Interactive.App.Connection.StdIoKernelConnector>;
+
 using CompositeDisposable = Pocket.CompositeDisposable;
 
 namespace Microsoft.DotNet.Interactive.App.Connection;
 
-public class StdIoKernelConnector : IKernelConnector, IDisposable
+public class StdIoKernelConnector : IKernelConnector
 {
     private KernelCommandAndEventReceiver? _receiver;
     private KernelCommandAndEventSender? _sender;
     private Process? _process;
+    private ProxyKernel? _rootProxyKernel;
+
     private RefCountDisposable? _refCountDisposable = null;
 
-    public StdIoKernelConnector(string[] command, Uri kernelHostUri, DirectoryInfo? workingDirectory = null)
+    public string[] Command { get; }
+
+    public string RootProxyKernelLocalName { get; }
+
+    public Uri KernelHostUri { get; }
+
+    public DirectoryInfo WorkingDirectory { get; }
```

Comment:
Does this need to be public?

Reply:
Looks like none of the existing public properties on `StdIoKernelConnector` (including `Command` and `KernelHostUri`) are used outside the class. I'll go ahead and convert these to `private` fields.

The newly added `ProcessId` property is the only one that needs to remain public so that tests can use it to validate process clean-up on disposal. Looks like we don't use `InternalsVisibleTo` in the repo - otherwise `ProcessId` could have been made internal for test use only. Let me know if you have any thoughts around leaving `ProcessId` as a public property...

Reply:
> Looks like we don't use InternalsVisibleTo in the repo - otherwise ProcessId could have been made internal for test use only. 

We don't use `InternalsVisibleTo` because it can't be scoped to specific methods, so over time more and more API surface will become coupled to the consuming assemblies. 

The fact that `ProcessId` is useful in tests is design feedback that it might be generally useful as part of the public API. I can imagine wanting to know, for example, which process I need to kill if the proxied kernels need to be aggressively shut down.

Reply:
There is some analyzer tech in the Roslyn repo that can be used to restrict IVT to specific namespaces within an assembly. If we really wanted to, I think it should be possible to scope that down to type / method level.

That said, I can also see how avoiding IVT entirely can be beneficial to avoid unintended coupling in tests. So long as we can keep the public API surface from getting bloated in unnatural ways (and I agree that `ProcessId` does not fall in this category)...

Reply:
> So long as we can keep the public API surface from getting bloated in unnatural ways (and I agree that ProcessId does not fall in this category)...

We make sure not to make things public just to test them. If all functionality is tested through intentional public APIs, then tests don't become too coupled to implementation details, and keeps tests in the zone where they better support refactoring.